packages/ui-navigation/src/AppNav/Item/props.ts

```diff
@@ -31,72 +31,77 @@ import type {
 } from '@instructure/shared-types'
 import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
 import type { Cursor } from '@instructure/ui-prop-types'
+import React from 'react'
 
 type AppNavItemOwnProps = {
-  renderLabel: React.ReactNode | ((...args: any[]) => any)
-  renderAfter?: React.ReactNode | ((...args: any[]) => any)
-  renderIcon?: React.ReactNode | ((...args: any[]) => any)
-  href?: string
-  onClick?: (...args: any[]) => any
-  isSelected?: boolean
-  elementRef?: (element: Element | null) => void
-  as?: AsElementType
-  cursor?: Cursor
-  isDisabled?: boolean
-}
-
-type PropKeys = keyof AppNavItemOwnProps
-
-type AllowedPropKeys = Readonly<Array<PropKeys>>
-
-type AppNavItemProps = AppNavItemOwnProps &
-  WithStyleProps<AppNavItemTheme, AppNavItemStyle> &
-  OtherHTMLAttributes<AppNavItemOwnProps>
-
-type AppNavItemStyle = ComponentStyle<'item' | 'label'>
-
-const propTypes: PropValidators<PropKeys> = {
   /**
    * The text to display. If the `icon` prop is used, label text must be wrapped
    * in `ScreenReaderContent`.
    */
-  renderLabel: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+  renderLabel: React.ReactNode | string | (() => React.ReactNode)
```

Comment:
The `string` type is not needed, it is included in `React.ReactNode`

```suggestion
  renderLabel: React.ReactNode | (() => React.ReactNode)
```